src/lib/dnssd/DnssdCache.h

```diff
@@ -60,18 +52,20 @@ class DnssdCache
 
         ResolvedNodeData * entry;
 
-        entry = FindPeerId(nodeData.mPeerId, currentTime);
+        entry = FindPeer(nodeData.mPeerId, currentTime);
         if (entry)
         {
             *entry = nodeData;
             return CHIP_NO_ERROR;
         }
 
-        VerifyOrReturnError(entry = findSlot(currentTime), CHIP_ERROR_TOO_MANY_KEYS);
+        if (mLookupTable.Allocated() >= CACHE_SIZE)
```

Comment:
Shouldn't this compare to mLookupTable.Capacity()?

Reply:
No we can't. When `CHIP_SYSTEM_CONFIG_POOL_USE_HEAP` is true, it returns `SIZE_MAX`. But here we want to limit the cache size to our given value.

Reply:
> But here we want to limit the cache size to our given value.

Do we, in the heap case?